src/main/java/com/ifmo/epampractice/daoimpl/UsersDaoImpl.java

```diff
@@ -0,0 +1,144 @@
+package com.ifmo.epampractice.daoimpl;
+
+import com.ifmo.epampractice.dao.BasicDaoInterface;
+import com.ifmo.epampractice.dao.DBConnectorInterface;
+import com.ifmo.epampractice.dao.DBConnectorPostgres;
+import com.ifmo.epampractice.dao.UsersDao;
+import com.ifmo.epampractice.entity.UsersEntity;
+import com.ifmo.epampractice.enums.UserRole;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UsersDaoImpl implements UsersDao {
+
+    private static final String GET_ALL_QUERY = "SELECT * FROM Users";
+    private static final String GET_QUERY = "SELECT * FROM Users WHERE ID = ?";
+    private static final String UPDATE_QUERY = "UPDATE Users SET (Email, Password_hash, Role, Name, Birth_date, " +
+            "Signup_date, Pass_id, Driving_license_id, Contact_phone, Address, Is_blocked, " +
+            "Reputation) = (?, ?, ?::e_role_users, ?, ?, ?, ?, ?, ?, ?, ?, ?) WHERE ID = ?";
+    private static final String DELETE_QUERY = "DELETE FROM Users WHERE ID = ?";
+    private static final String INSERT_QUERY = "INSERT INTO Users (Email, Password_hash, Role, Name, Birth_date, " +
+            "Signup_date, Pass_id, Driving_license_id, Contact_phone, Address, Is_blocked, " +
+            "Reputation) VALUES (?, ?, ?::e_role_users, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+
+    @Override
+    public List<UsersEntity> getAll() {
+        List<UsersEntity> users = new ArrayList<>();
+
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+        try (Connection connection = dbConnector.getConnection()) {
+            Statement statement = connection.createStatement(
+                    ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+            ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY);
+            while (resultSet.next()) {
+                users.add(entityFromResultSet(resultSet));
+            }
+            return users;
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+            //TODO: use logger when it will be set up.
+            return users;
+        }
+    }
+
+    @Override
+    public UsersEntity get(int id) {
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+        try (Connection connection = dbConnector.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(GET_QUERY,
+                    ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+            statement.setInt(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
```

Comment:
is not the best way, the thing is that `!resultSet.next()` makes reading more confusing than just `resultSet.next()`
try:
```
            if (resultSet.next()) {
                return entityFromResultSet(resultSet);
            }
        } catch (SQLException e) {
          ...
        }
        return null;
```